docs/guidelines/custom-elements-warp/index.md

```diff
@@ -0,0 +1,242 @@
+
+## Usage in an app
+
+### Option 1: Client side only
+
+Import the base class from the Eik server directly
+
+```jsx
+import WarpElement from "https://assets.finn.no/pkg/@warp-ds/elements-core/v0/element.js";
+import { html } from "lit";
+```
+
+then, extend the base class and include the base classes styles in your component
+
+```jsx
+class MyElement extends WarpElement {
+  static styles = [WarpElement.styles];
+
+  render() {
+    return html`<div class="...use warp styles...">hello world</div>`;
+  }
+}
+```
+
+### Option 2: Server side only
+
+Install the package
+
+```jsx
+npm install @warp-ds/elements-core
+```
+
+next, import the base class from the package
+
+```jsx
+import WarpElement from "@warp-ds/elements-core";
+import { html } from "lit";
+```
+
+Finally, extend the base class and include the base classes styles in your component
+
+```jsx
+class MyElement extends WarpElement {
+    static styles = [WarpElement.styles];
+
+  render() {
+    return html`<div class="...use warp styles...">hello world</div>`;
+  }
+}
+```
+
+### Option 3: Server side with client side hydration
+
+The starting point for this is identical to the server side only option described above.
+
+After this, we need to add a build step for the client side that uses Eik import mapping to map the package to a URL on the Eik server for us.
+
+**Using Esbuild**
+
+Firstly, include the correct import map file in your `eik.json` file
+
+```json
+{
+	"name": "myproject",
+    "version": "1.0.0",
+	"files": "./dist",
+    "import-map": ["https://assets.finn.no/map/custom-elements/v2"]
+}
+```
+
+Next, install build dependencies
+
+```bash
+npm install --save-dev @eik/esbuild-plugin esbuild
+```
+
+Create a build file with the following contents
+
+```jsx
+// build.js
+
+import * as eik from '@eik/esbuild-plugin';
+import esbuild from 'esbuild';
+
+await eik.load();
+
+await esbuild.build({
+    entryPoints: ['./src/app.js'],
+    outdir: "./dist",
+    bundle: true,
+    format: 'esm',
+    target: ['es2021'],
+    plugins: [eik.plugin()],
+});
+```
+
+Next, add an npm run scripts to `package.json`
+
+```json
+{
+  ...
+	"scripts": {
+    ...
+    "build": "node build.js"
+    ...
+  }
+  ...
+}
+```
+
+And finally, run the script to produce a build with references to Lit and Warp Element replaced with references to URLs on the Eik server
+
+```bash
+npm run build
+```
+
+## Publishing components
+
+If you are creating a component or component library that is to be consumed by other teams you will need to create and provide components your components and include instructions on how to setup import mapping.
+
```

Comment:
This sentence is a bit confusing, looks like duplication of components. Maybe you can remove the `your components` part? :)